### PR TITLE
Reverts #1557

### PR DESCRIPTION
**Pull request by DhairyaLGandhi**
Reverts #1557

### CODE REVIEW

**Comment by DhairyaLGandhi** [comment]
Closes #1560 #1559 

cc @darsnack @CarloLucibello

**Review by CarloLucibello — APPROVED** [review]

File: test/layers/basic.jl [hunk 1]
```diff
@@ -40,17 +40,16 @@ import Flux: activations
       @test Dense(rand(Float16, 100,10), true).bias isa Vector{Float16}  # creates matching type
       @test_skip Dense(rand(Float16, 100,10), rand(100)).bias isa Vector{Float16}  # converts to match
 
-      @test_skip Dense(3,4; init=Base.randn, bias=true).bias isa Vector{Float64}
+      @test Dense(3,4; init=Base.randn, bias=true).bias isa Vector{Float64}
```
Comment (darsnack) [inline test/layers/basic.jl:43]: This should be a `@test_skip` right? The correct `eltype` passing is an accident here (cause `bias = true` uses `zeros` which happens to be `Float64`).

**Review by darsnack — CHANGES_REQUESTED** [review]
Believe this is correct other than the one test

File: test/layers/basic.jl (hunk 1 above)
Comment (DhairyaLGandhi) [reply test/layers/basic.jl:43]: No. This was in #1558, and it uses `Flux.zeros` (not `Base.zeros`) which produces `Float32` by default.

```julia
julia> Flux.zeros(4)
4-element Array{Float32,1}:
 0.0
 0.0
 0.0
 0.0
```

I agree that the test isn't very explicit about its intent.

Comment (darsnack) [reply test/layers/basic.jl:43]: Ah I see, okay we'll address it separately. Thanks!

Comment (CarloLucibello) [reply test/layers/basic.jl:43]: Reverting here is good, we should produce float64 bias when the weights produced are float64

**Comment by darsnack** [comment]
Okay so do I need to hit approve or is `bors` okay?

File: test/layers/basic.jl (hunk 1 above)
Comment (DhairyaLGandhi) [reply test/layers/basic.jl:43]: > we should produce float64 bias when the weights produced are float64

No
We should do as we are told by the user. Flux is meant to be hackable. While it is valid to have defaults, it is required to have overrides at different levels.

**Pull request merged**